tests/rust-plugins/swc_internal_plugin/src/lib.rs

```diff
@@ -29,28 +30,27 @@ impl VisitMut for ConsoleOutputReplacer {
 // - this is not complete signature, need to take config_json
 // - swc_plugin macro to support ergonomic interfaces
 // - better developer experience: println / dbg!() doesn't emit anything
-// - typed json config instead of str, which requires additional deserialization
-pub fn process(ast_ptr: *mut u8, len: u32) -> (i32, i32) {
+pub fn process(ast_ptr: *mut u8, len: i32) -> (i32, i32) {
     // Read raw serialized bytes from wasm's memory space. Host (SWC) should
-    // allocated memory, copy bytes and pass ptr to it.
+    // allocate memory, copy bytes and pass ptr to plugin.
     let raw_serialized_bytes =
         unsafe { std::slice::from_raw_parts(ast_ptr, len.try_into().unwrap()) };
 
-    // Reconstruct AlignedVec from raw bytes
-    let mut serialized_program = AlignedVec::with_capacity(len.try_into().unwrap());
-    serialized_program.extend_from_slice(raw_serialized_bytes);
+    // Reconstruct SerializedProgram from raw bytes
+    let serialized_program = SerializedProgram::new(raw_serialized_bytes.to_vec(), len);
```

Comment:
`to_vec` clones memory. Is this cloning required?

Reply:
Original code also copy memory, though.

Reply:
probably..not? Since this is plugin code for the testing, I honestly choose some kind of shortcut here.

Reply:
Oh, you are correct. It does not matter I think.
We need to think about once we provide a macro for plugins.